crates/fuzzing/src/oracles/diff_wasmi.rs

```diff
@@ -141,35 +129,32 @@ impl DiffInstance for WasmiInstance {
         arguments: &[DiffValue],
         result_tys: &[DiffValueType],
     ) -> Result<Option<Vec<DiffValue>>> {
-        let function = match self
+        let function = self
             .instance
             .get_export(&self.store, function_name)
-            .unwrap()
-        {
-            wasmi::Extern::Func(f) => f,
-            _ => unreachable!(),
-        };
+            .and_then(wasmi::Extern::into_func)
+            .unwrap();
         let arguments: Vec<_> = arguments.iter().map(|x| x.into()).collect();
         let mut results = vec![wasmi::core::Value::I32(0); result_tys.len()];
         function
             .call(&mut self.store, &arguments, &mut results)
             .context("wasmi function trap")?;
-        Ok(Some(results.into_iter().map(|x| x.into()).collect()))
+        Ok(Some(results.into_iter().map(Into::into).collect()))
     }
 
     fn get_global(&mut self, name: &str, _ty: DiffValueType) -> Option<DiffValue> {
-        match self.instance.get_export(&self.store, name).unwrap() {
-            wasmi::Extern::Global(g) => Some(g.get(&self.store).into()),
-            _ => unreachable!(),
-        }
+        self.instance
+            .get_export(&self.store, name)
+            .and_then(wasmi::Extern::into_global)
+            .map(|global| global.get(&self.store).into())
```

Comment:
Could the `.unwrap()` be retained here and on `memory`? If the export isn't present or is of a different type that's a bug in the fuzz harness which we need to fix and the `.unwrap()` helps uncoveri t.

Reply:
yeah I just noticed that this `unwrap` might have been intentional for exactly that purpose. I will try to fix it.

Reply:
under what circumstances is `get_global` supposed to return `Some` value?

Reply:
Something like
````rust
self.instance
    .get_export(&self.store, name)
    .and_then(wasmi::Extern::into_global)
    .map(|global| global.get(&self.store))
    .map(DiffValue::from)
    .unwrap()
    .into()
```
Or would you rather have the original code back?

Reply:
This is intended for JS where JS can't get the value of an exported `v128` global, for example. So only for situations where the engine for some reason is unable to retrieve the value. For `wasmi` though it should be able to retrieve the value in all situations.